rs/replicated_state/src/replicated_state.rs

```diff
@@ -929,6 +929,43 @@ impl ReplicatedState {
         timed_out_messages_count
     }
 
+    /// Times out all callbacks with expired deadlines (given the state time) that
+    /// have not already been timed out. Returns the number of timed out callbacks.
+    ///
+    /// See `CanisterQueues::time_out_callbacks` for further details.
+    pub fn time_out_callbacks(&mut self) -> (usize, Vec<StateError>) {
+        let current_time = CoarseTime::floor(self.metadata.time());
+        // Because the borrow checker requires us to remove each canister before
```

Comment:
IIUC we're doing this twice now, in `time_out_messages` and `time_out_callbacks`, but I guess the code is cleaner this way. Maybe worth a comment?

But yeah, would be nice to have something like `split_at_mut` for BTreeMaps.

Reply:
The problem we have is that we want to mutate a given `CanisterState` while being able to query the keys of `canister_states` (to decide whether a canister is local or not).

I'll look into other ways of deciding whether a canister is local or not. One option is to maintain a `BTreeSet` in parallel with `canister_states` for this purpose (but because `canister_states` is public, this is likely problematic). The other option is to rely on the routing table instead.